azkaban-common/src/main/java/azkaban/executor/ExecutableJobInfo.java

```diff
@@ -122,6 +116,8 @@ private void parseFlowId() {
 
       this.jobPath.add(pair);
     }
+
+    this.immediateFlowId = this.jobPath.get(this.jobPath.size() - 1).getSecond();
```

Comment:
Right now Azkaban doesn't allow the user to upload a project with no flows so jobPath is guaranteed to have at least one element, but since changes to this behavior would mean a contract violation I believe it would make sense to rewrite it like this:

Assert.assertFalse("There most be at least one flow in project", this.jobPath.isEmpty());
this.immediateFlowId = this.jobPath.get(this.jobPath.size() - 1).getSecond();

For reference:
https://stackoverflow.com/questions/2440984/assert-a-good-practice-or-not